src/main/java/com/ibm/eventstreams/connect/mqsource/MQSourceConnector.java

```diff
@@ -218,104 +223,264 @@ public void stop() {
-                      CONFIG_DOCUMENTATION_MQ_MESSAGE_BODY_JMS, CONFIG_GROUP_MQ, 10, Width.SHORT,
-                      CONFIG_DISPLAY_MQ_MESSAGE_BODY_JMS);
-
-        config.define(CONFIG_NAME_MQ_RECORD_BUILDER_KEY_HEADER, Type.STRING, null, Importance.MEDIUM,
-                      CONFIG_DOCUMENTATION_MQ_RECORD_BUILDER_KEY_HEADER, CONFIG_GROUP_MQ, 11, Width.MEDIUM,
-                      CONFIG_DISPLAY_MQ_RECORD_BUILDER_KEY_HEADER);
-
-        config.define(CONFIG_NAME_MQ_SSL_CIPHER_SUITE, Type.STRING, null, Importance.MEDIUM,
-                      CONFIG_DOCUMENTATION_MQ_SSL_CIPHER_SUITE, CONFIG_GROUP_MQ, 12, Width.MEDIUM,
-                      CONFIG_DISPLAY_MQ_SSL_CIPHER_SUITE);
-
-        config.define(CONFIG_NAME_MQ_SSL_PEER_NAME, Type.STRING, null, Importance.MEDIUM,
-                      CONFIG_DOCUMENTATION_MQ_SSL_PEER_NAME, CONFIG_GROUP_MQ, 13, Width.MEDIUM,
-                      CONFIG_DISPLAY_MQ_SSL_PEER_NAME);
-
-        config.define(CONFIG_NAME_MQ_SSL_KEYSTORE_LOCATION, Type.STRING, null, Importance.MEDIUM,
-                      CONFIG_DOCUMENTATION_MQ_SSL_KEYSTORE_LOCATION, CONFIG_GROUP_MQ, 14, Width.MEDIUM,
-                      CONFIG_DISPLAY_MQ_SSL_KEYSTORE_LOCATION);
-
-        config.define(CONFIG_NAME_MQ_SSL_KEYSTORE_PASSWORD, Type.PASSWORD, null, Importance.MEDIUM,
-                      CONFIG_DOCUMENTATION_MQ_SSL_KEYSTORE_PASSWORD, CONFIG_GROUP_MQ, 15, Width.MEDIUM,
-                      CONFIG_DISPLAY_MQ_SSL_KEYSTORE_PASSWORD);
-
-        config.define(CONFIG_NAME_MQ_SSL_TRUSTSTORE_LOCATION, Type.STRING, null, Importance.MEDIUM,
-                      CONFIG_DOCUMENTATION_MQ_SSL_TRUSTSTORE_LOCATION, CONFIG_GROUP_MQ, 16, Width.MEDIUM,
-                      CONFIG_DISPLAY_MQ_SSL_TRUSTSTORE_LOCATION);
-
-        config.define(CONFIG_NAME_MQ_SSL_TRUSTSTORE_PASSWORD, Type.PASSWORD, null, Importance.MEDIUM,
-                      CONFIG_DOCUMENTATION_MQ_SSL_TRUSTSTORE_PASSWORD, CONFIG_GROUP_MQ, 17, Width.MEDIUM,
-                      CONFIG_DISPLAY_MQ_SSL_TRUSTSTORE_PASSWORD);
-
-        config.define(CONFIG_NAME_MQ_BATCH_SIZE, Type.INT, CONFIG_VALUE_MQ_BATCH_SIZE_DEFAULT,
-                      ConfigDef.Range.atLeast(CONFIG_VALUE_MQ_BATCH_SIZE_MINIMUM), Importance.LOW,
-                      CONFIG_DOCUMENTATION_MQ_BATCH_SIZE, CONFIG_GROUP_MQ, 18, Width.MEDIUM,
-                      CONFIG_DISPLAY_MQ_BATCH_SIZE);
+        return CONFIGDEF;
+    }
 
-        config.define(CONFIG_NAME_MQ_MESSAGE_MQMD_READ, Type.BOOLEAN, Boolean.FALSE, Importance.LOW,
-                      CONFIG_DOCUMENTATION_MQ_MESSAGE_MQMD_READ, CONFIG_GROUP_MQ, 19, Width.SHORT,
-                      CONFIG_DISPLAY_MQ_MESSAGE_MQMD_READ);
+    /** Null validator - indicates that any value is acceptable for this config option. */
+    private static final ConfigDef.Validator ANY = null;
+
+    static {
+        CONFIGDEF = new ConfigDef();
+
+        CONFIGDEF.define(CONFIG_NAME_MQ_QUEUE_MANAGER,
```

Comment:
Is `CONFIG_NAME` useful as a prefix? could we simply use to the `DOCUMENTATION` and `MESSAGE` prefixes for the corresponding strings? and remove the `CONFIG_NAME` prefix altogether?

Reply:
same sort of thing here, I'm afraid - as these are public statics in a project that users are encouraged to develop against, I feel like these can't be altered without a major semantic version change

I agree that they're not ideal as/where they are though